projects/plugins/jetpack/modules/theme-tools/site-logo.php

```diff
@@ -22,13 +22,18 @@
  * Activate the Site Logo plugin.
  *
  * @uses current_theme_supports()
- * @since 3.2
+ * @since 3.2.0
+ * @since 9.9.0 Uses Core site_logo option format universally.
  */
 function site_logo_init() {
-	// For transferring existing site logo from Jetpack -> Core
-	if ( current_theme_supports( 'custom-logo' ) && ! get_theme_mod( 'custom_logo' ) && $jp_logo = get_option( 'site_logo' ) ) {
-		set_theme_mod( 'custom_logo', $jp_logo['id'] );
-		delete_option( 'site_logo' );
+	// For transferring existing site logo from Jetpack -> Core.
+	if ( current_theme_supports( 'custom-logo' ) && ! get_theme_mod( 'custom_logo' ) && get_option( 'site_logo' ) ) {
+		$jp_logo = get_option( 'site_logo' );
+		// Is the Site Logo in the old Jetpack format?
+		if ( ! empty( $jp_logo['id'] ) ) {
+			set_theme_mod( 'custom_logo', $jp_logo['id'] );
+			update_option( 'site_logo', $jp_logo['id'] );
```

Comment:
I wonder if we should have a comment in here explaining that this `site_logo` isn't Jetpack anymore?

Reply:
Makes sense. a256266a6